internal/collector/telemetry/telemetry.go

```diff
@@ -49,6 +50,11 @@ func Flags(flags *flag.FlagSet) {
 		"add-instance-id",
 		true,
 		"Flag to control the addition of 'service.instance.id' to the collector metrics.")
+
+	addCollectorVersionPtr = flags.Bool(
+		"add-collector-version",
+		true,
+		"Flag to control the addition of 'service.version' to the collector metrics.")
 }
```

Comment:
Please do not add more flags, see https://github.com/open-telemetry/opentelemetry-collector/issues/3957 and https://github.com/open-telemetry/opentelemetry-collector/issues/3843

Reply:
I'll wait until https://github.com/open-telemetry/opentelemetry-collector/milestone/21 is done.